File: sample-csrf/src/main/java/blog/csrf/demo/SampleController.java
package blog.csrf.demo;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class SampleController {

    @GetMapping("/")
    public void sampleGet(){
    }

    @PostMapping("/")
    public void samplePost(){
    }

}

File: kotlin-junit5/src/main/kotlin/com/example/kotlinjunit5/sample/SampleController.kt
package com.example.kotlinjunit5.sample

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/sample")
class SampleController(
        private val aRepository: ARepository,
        private val bRepository: BRepository
) {
    @GetMapping
    fun transaction() {
        aRepository.save(A("A"))
        bRepository.save(B("B"))
    }
}